src/main/groovy/com/deploygate/gradle/plugins/DeployGatePlugin.groovy

```diff
@@ -54,108 +55,88 @@ class DeployGatePlugin implements Plugin<Project> {
 
         GradleCompat.init(project)
 
-        project.tasks.register(Constants.LOGIN_TASK_NAME, LoginTask) { task ->
-            task.description = "Check the configured credentials and launch the authentication flow if they are not enough."
+        def loginTask = project.tasks.register(Constants.LOGIN_TASK_NAME, LoginTask) { task ->
+            DeployGateExtension deploygate = project.deploygate
 
-            task.group = Constants.TASK_GROUP_NAME
-            task.deployGateExtension = project.deploygate
+            task.deployGateExtension = deploygate
+            task.appOwnerName.set(deploygate.appOwnerName)
+            task.apiToken.set(deploygate.apiToken)
         }
 
         project.tasks.register(Constants.LOGOUT_TASK_NAME, LogoutTask) { task ->
-            task.description = "Remove the local persisted credentials."
-
-            task.group = Constants.TASK_GROUP_NAME
             task.credentialStore = project.deploygate.credentialStore
         }
 
-        project.tasks.register(DeployGateTaskFactory.SUFFIX_APK_TASK_NAME, DefaultTask).configure { task ->
+        project.tasks.register(Constants.SUFFIX_APK_TASK_NAME, DefaultTask).configure { task ->
             task.description = "Execute all custom upload-apk tasks"
 
-            task.group = DeployGateTaskFactory.GROUP_NAME
+            task.group = Constants.TASK_GROUP_NAME
         }
 
-        project.tasks.register(DeployGateTaskFactory.SUFFIX_AAB_TASK_NAME, DefaultTask).configure { task ->
+        project.tasks.register(Constants.SUFFIX_AAB_TASK_NAME, DefaultTask).configure { task ->
             task.description = "Execute all custom upload-aab tasks"
 
-            task.group = DeployGateTaskFactory.GROUP_NAME
+            task.group = Constants.TASK_GROUP_NAME
         }
 
-        project.deploygate.deployments.configureEach { NamedDeployment d ->
-            project.tasks.named(DeployGateTaskFactory.SUFFIX_APK_TASK_NAME).configure { task ->
-                task.dependsOn(DeployGateTaskFactory.uploadApkTaskName(d.name))
+        project.deploygate.deployments.configureEach { NamedDeployment deployment ->
+            project.tasks.named(Constants.SUFFIX_APK_TASK_NAME).configure { task ->
+                task.dependsOn(Constants.uploadApkTaskName(deployment.name))
             }
 
-            project.tasks.named(DeployGateTaskFactory.SUFFIX_AAB_TASK_NAME).configure { task ->
-                task.dependsOn(DeployGateTaskFactory.uploadAabTaskName(d.name))
+            project.tasks.named(Constants.SUFFIX_AAB_TASK_NAME).configure { task ->
+                task.dependsOn(Constants.uploadAabTaskName(deployment.name))
             }
 
-            project.tasks.register(DeployGateTaskFactory.uploadApkTaskName(d.name), UploadApkTask) { task ->
-                final NamedDeployment deployment = project.deploygate.findDeploymentByName(d.name)
```

Comment:
Created a DeploymentConfiguration specified for Gradle inputs so don't need to re-query NamedDeployment